praw/models/reddit/subreddit.py

```diff
@@ -2278,6 +2361,74 @@ def upload(self, name, image_path):
         return self._upload_image(image_path,
                                   {'name': name, 'upload_type': 'img'})
 
+    def upload_banner(self, image_path):
+        """Upload an image to be used as the Subreddit's (Reddit redesign) banner image.
+
+        :param image_path: A path to a jpeg or png image.
+        :returns: The url for the uploaded file in S3.
```

Comment:
Why return the S3 url of the uploaded file? My understanding of the banner system is limited, but is there perhaps some useful response from `self.subreddit._reddit.patch(style_url, structured_style)` that could be returned?

This comment is likely related to my comment above regarding the functionality and return value of `_upload_style_asset`.

Same question/comment for `upload_banner_additional_image` and `upload_banner_hover_image`.

Reply:
Unfortunately, the patch call doesn't really include anything useful (that I can see) in the return. I'm not sure what a really good return for these upload_*_image methods would be other than a self reference to the SubredditStylesheet to allow for chaining function calls.

I suppose with an additional call, they could return the post-update structured_style dict so callers could maybe validate what was done?  Or I could just make _fetch_structured_style() public.

Reply:
In my opinion, a return value of `None` is safer than returning the S3 url. My understanding is that the S3 URL is temporary, and the subsequent call to `_update_structured_styles` moves the location of the file, invalidating the S3 URL. (see [this section](https://www.reddit.com/dev/api#POST_api_v1_{subreddit}_emoji_asset_upload_s3.json) of Reddit API docs, regarding S3 in emoji upload). This seems to be true in my testing — trying to open the URL returned by this method results in 403 forbidden.

If there isn't useful information to return, that's alright. But it doesn't seem like this return value makes sense.

Again, same for the other upload methods.